src/event-bridge/event-pattern/synth.ts

```diff
@@ -125,16 +125,12 @@ export const synthesizeEventPattern = (
  * https://github.com/functionless/functionless/issues/37#issuecomment-1066313146
  */
 export const synthesizePatternDocument = (
-  predicate: FunctionDecl | Err | unknown
+  predicate: RulePredicateFunction<any, any> | FunctionLike | Err | undefined
 ): PatternDocument => {
-  if (isErr(predicate)) {
-    throw predicate.error;
-  } else if (!isFunctionDecl(predicate)) {
-    throw Error(
-      "Expected parameter to synthesizeEventPattern to be compiled by functionless."
-    );
-  }
-  const [eventDecl = undefined] = (<FunctionDecl>predicate).parameters;
+  // @ts-ignore
+  const func = validateFunctionLike(predicate, "synthesizeEventPattern");
```

Comment:
Instead of `ts-ignore`, should we just type the predicate as `unknown as FunctionLike`? That way the rest of the type checking will work.

Reply:
This `ts-ignore` was not required